torch/_higher_order_ops/cond.py

```diff
@@ -28,11 +30,51 @@
 )
 
 
+@contextmanager
+def _set_compilation_env():
+    _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    _old_capture_dyn_ops = torch._dynamo.config.capture_dynamic_output_shape_ops
+    _old_capture_scalar = torch._dynamo.config.capture_scalar_outputs
+    _old_size_limit = torch._dynamo.config.cache_size_limit
+    _old_acc_size_limit = torch._dynamo.config.accumulated_cache_size_limit
+
+    try:
+        # We need to turn off the is_fx_tracing_flag temporarily
+        # Remove it once we are confident fx tracing works with dynamo
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
+        torch._dynamo.config.capture_dynamic_output_shape_ops = True
+        torch._dynamo.config.capture_scalar_outputs = True
+        # We need to enlarge the dynamo cache size limit because
+        # cond's frame is expected to be hitted with different inputs frequently
+        torch._dynamo.config.cache_size_limit = 256
+        torch._dynamo.config.accumulated_cache_size_limit = 256
```

Comment:
Each cond call is going to check against the cache, and since we didn't reset dynamo between tests, we end up accumulating a lot of caches for that cond operation.

May be we should mannuly call dynamo.reset() between the tests?